daemon/inertiad/project/data.go

```diff
@@ -138,6 +140,49 @@ func (c *DeploymentDataManager) GetEnvVariables(decrypt bool) ([]string, error)
 	return envs, err
 }
 
+// AddProjectBuildData stores and tracks metadata from successful builds
+// TODO: definitely need to add more context to errors
+func (c *DeploymentDataManager) AddProjectBuildData(projectName string, mdata DeploymentMetadata) error {
+	// encode metadata so it can be stored as byte array
+	encodedMdata, err := json.Marshal(mdata)
+	if err != nil {
+		return err
+	}
+	return c.db.Update(func(tx *bolt.Tx) error {
+		deployedProjectsBucket := tx.Bucket(deployedProjectsBucket)
+		// if bkt with project name doesnt exist create new bkt, otherwise update existing bucket
+		if projectBkt := deployedProjectsBucket.Bucket([]byte(projectName)); projectBkt == nil {
+			projectBkt, err := deployedProjectsBucket.CreateBucket([]byte(projectName))
+			if err != nil {
+				return err
+			}
+
+			if err := projectBkt.Put([]byte(time.Now().String()), encodedMdata); err != nil {
+				return err
+			}
+		} else {
+			if err := c.UpdateProjectBuildData(projectName, encodedMdata); err != nil {
+				return err
+			}
+		}
+		return nil
+	})
+}
+
+// UpdateProjectBuildData updates existing project bkt with recent build's metadata
+func (c *DeploymentDataManager) UpdateProjectBuildData(projectName string, mdata []byte) error {
```

Comment:
this should either:

a) have the same signature as `AddProjectBuildData(projectName string, mdata DeploymentMetadata) error`
b) be removed, in favor of an option (`overwrite bool` for example) in `AddProjectBuildData`, or just always let `AddProjectBuildData` overwrite existing data